project/AppDependencies.scala

```diff
@@ -1,38 +1,32 @@
-
 import play.core.PlayVersion.current
-import sbt._
+import sbt.*
 
 object AppDependencies {
 
-  val compile: Seq[ModuleID] = Seq(
-    "uk.gov.hmrc" %% "bootstrap-frontend-play-28" % "7.1.0",
-    "uk.gov.hmrc" %% "bootstrap-backend-play-28" % "7.1.0",
-    "uk.gov.hmrc.mongo" %% "hmrc-mongo-play-28" % "0.71.0",
-    "uk.gov.hmrc" %% "play-frontend-hmrc" % "3.23.0-play-28",
-    "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.13.3"
-  )
+  private val bootstrapPlayVersion = "7.14.0" // later version breaks the play's binding in this project as of 2023.05 up to version 7.15.0
+  private val mongoVersion = "0.74.0"
 
-  def commonTestDependencies(scope: Configuration): Seq[ModuleID] = Seq(
-    "org.scalatestplus.play" %% "scalatestplus-play" % "5.1.0" % scope,
-    "org.scalatest" %% "scalatest" % "3.1.1" % scope,
-    "org.jsoup" % "jsoup" % "1.14.1" % scope,
-    "com.typesafe.play" %% "play-test" % current % scope,
-    "com.vladsch.flexmark" % "flexmark-all" % "0.36.8" % scope,
-    "uk.gov.hmrc.mongo" %% "hmrc-mongo-test-play-28" % "0.71.0" % scope
+  val compile: Seq[ModuleID] = Seq(
+    "uk.gov.hmrc"       %% "bootstrap-frontend-play-28" % bootstrapPlayVersion,
+    "uk.gov.hmrc"       %% "bootstrap-backend-play-28"  % bootstrapPlayVersion,
+    "uk.gov.hmrc.mongo" %% "hmrc-mongo-play-28"         % mongoVersion,
+    "uk.gov.hmrc"       %% "play-frontend-hmrc"         % "6.8.0-play-28", // later version breaks nonce-CSP as of 2023.05 up to version 7.7.0
+    "com.fasterxml.jackson.module" %% "jackson-module-scala" % "2.14.2"
   )
 
   val test: Seq[ModuleID] = Seq(
-    // Bootstrap import needed to access stubMessagesControllerComponents in unit tests
-    "uk.gov.hmrc" %% "bootstrap-frontend-play-28" % "7.1.0",
-    "uk.gov.hmrc" %% "bootstrap-backend-play-28" % "7.1.0",
-    "org.mockito" % "mockito-core" % "3.11.2" % Test,
-    "org.scalatestplus" %% "mockito-3-4" % "3.2.10.0" % Test
-  ) ++ commonTestDependencies(Test)
-
-  val it: Seq[ModuleID] = Seq(
-    "com.github.tomakehurst" % "wiremock-jre8" % "2.29.1" % IntegrationTest
-  ) ++ commonTestDependencies(IntegrationTest)
+    "org.mockito"             % "mockito-core"            % "5.2.0"              % "test",
+    "org.scalatestplus"      %% "mockito-3-4"             % "3.2.10.0"           % "test",
+    "org.scalatestplus.play" %% "scalatestplus-play"      % "5.1.0"              % "test,it",
+    "org.scalatest"          %% "scalatest"               % "3.2.15"             % "test,it",
```

Comment:
Try dropping `"org.scalatest" %% "scalatest" % "3.2.15"`. I believe it's pulled in by Bootstrap.

Reply:
compiler is not happy about it. I think it is needed to make sure we are using the right version of scalatest and flexmark. Otherwise, compiler starts complaining about `java.lang.NoClassDefFoundError: com/vladsch/flexmark/util/ast/Node` and scalatest and flexmark versions fallback to older versions